// query-param/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.queryParam = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _extractors = require('./extractors');

var _validators = require('./validators');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var getBaseProperties = function getBaseProperties(_ref) {
  var url = _ref.url;
  var valid = _ref.valid;
  return { url: url, valid: valid };
};

var getExtractedProperties = function getExtractedProperties(url) {
  return (0, _extends3.default)({
    host: (0, _extractors.extractHost)(url),
    hostname: (0, _extractors.extractHost)(url) + ':' + (0, _extractors.extractPort)(url),
    port: (0, _extractors.extractPort)(url),
    protocol: (0, _extractors.extractProtocol)(url)
  }, getBaseProperties({ url: url, valid: true }));
};

var queryParam = exports.queryParam = function queryParam(url) {
  if (!(0, _validators.isValidUrl)(url)) {
    return getBaseProperties({ url: url, valid: false });
  }
  return getExtractedProperties(url);
};
